ggml/src/ggml-cpu/arch/x86/quants.c

```diff
@@ -1016,20 +964,14 @@ void ggml_vec_dot_q8_0_q8_0(int n, float * GGML_RESTRICT s, size_t bs, const voi
         accum = _mm256_add_ps(_mm256_mul_ps(deltas, p), accum);
     }
 
-    sumf = hsum_float_8(accum);
-
+    *s = hsum_float_8(accum);
+#else
+    UNUSED(nb);
+    UNUSED(ib);
+    UNUSED(x);
+    UNUSED(y);
+    ggml_vec_dot_q8_0_q8_0_generic(n, s, bs, vx, bx, vy, by, nrc);
 #endif
-    for (; ib < nb; ++ib) {
```

Comment:
This loop is necessary to handle cases where the number of blocks is odd. There may be other cases.